components/support/utils/src/main/java/mozilla/components/support/utils/WebURLFinder.kt

```diff
@@ -75,7 +75,7 @@ class WebURLFinder {
     companion object {
         // Taken from mozilla.components.support.utils.URLStringUtils. See documentation
         // there for a complete description.
-        private const val autolinkWebUrlPattern = "(\\w+-)*\\w+(://[/]*|:|\\.)(\\w+-)*\\w+([\\S&&[^\\w-]]\\S*)?"
+        private const val autolinkWebUrlPattern = "^\\s*(\\w+-+)*\\w+(://[/]*|:|\\.)(\\w+-+)*\\w+([\\S&&[^\\w-]]\\S*)?\\s*\$"
```

Comment:
If we can remove the clipboard usages (i.e. my other comment), that leaves just the `TabIntentProcessor.processSendIntent` usage (and maybe +1 clipboard usage). I looked at it and it seems like it's incorrect: we use `WebURLFinder` to figure out whether or not we should load a URL or search. However, we have the same problem when the user enters a URL: why is the problem solved twice, in two different ways? It feels like we can change the `processSendIntent` code to be the same as the "user entered url" code and remove this `WebURLFinder` usage. That also benefits us: the `WebURLFinder` usage is known to have performance problems.

**Can you file a follow-up for fenix/ac to look at about this issue as a general code health issue** (and tag it eng:perf too because it'd fix the perf)? Let me know if the problem doesn't make sense.